kafka-rest/src/main/java/io/confluent/kafkarest/controllers/AbstractConsumerLagManager.java

```diff
@@ -0,0 +1,158 @@
+import io.confluent.kafkarest.entities.Partition;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.ws.rs.NotFoundException;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsOptions;
+import org.apache.kafka.clients.admin.ListOffsetsOptions;
+import org.apache.kafka.clients.admin.ListOffsetsResult.ListOffsetsResultInfo;
+import org.apache.kafka.clients.admin.OffsetSpec;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.IsolationLevel;
+import org.apache.kafka.common.TopicPartition;
+
+abstract class AbstractConsumerLagManager {
+
+  private final Admin kafkaAdminClient;
+
+  @Inject
+  AbstractConsumerLagManager(
+      Admin kafkaAdminClient
+  ) {
+    this.kafkaAdminClient = kafkaAdminClient;
+  }
+
+  protected final Map<TopicPartition, MemberId> getMemberIds(
+      ConsumerGroup consumerGroup
+  ) {
+    Map<TopicPartition, MemberId> tpMemberIds = new HashMap<>();
+    for (Consumer consumer: consumerGroup.getConsumers()) {
+      for (Partition partition : consumer.getAssignedPartitions()) {
+        MemberId memberId =
+            MemberId.builder()
+                .setConsumerId(consumer.getConsumerId())
+                .setClientId(consumer.getClientId())
+                .setInstanceId(consumer.getInstanceId().orElse(null))
+                .build();
+        TopicPartition topicPartition =
+            new TopicPartition(partition.getTopicName(), partition.getPartitionId());
+        tpMemberIds.put(topicPartition, memberId);
+      }
+    }
+    return tpMemberIds;
+  }
+
+  protected final CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> getCurrentOffsets(
+      String consumerGroupId
+  ) {
+    return KafkaFutures.toCompletableFuture(
+        kafkaAdminClient.listConsumerGroupOffsets(
+            consumerGroupId, new ListConsumerGroupOffsetsOptions())
+            .partitionsToOffsetAndMetadata());
+  }
+
+  protected final CompletableFuture<Map<TopicPartition, ListOffsetsResultInfo>> getLatestOffsets(
+      Map<TopicPartition, OffsetAndMetadata> currentOffsets
+  ) {
+    IsolationLevel isolationLevel = IsolationLevel.READ_COMMITTED;
+    Map<TopicPartition, OffsetSpec> latestOffsetSpecs =
+        currentOffsets.keySet()
+            .stream()
+            .collect(
+                Collectors.toMap(Function.identity(), topicPartition -> OffsetSpec.latest()));
+    return KafkaFutures.toCompletableFuture(
+        kafkaAdminClient.listOffsets(
+            latestOffsetSpecs,
+            new ListOffsetsOptions(isolationLevel)).all());
+  }
+
+  protected final long getCurrentOffset(Map<TopicPartition, OffsetAndMetadata> map, TopicPartition topicPartition) {
```

Comment:
Rename `map` to something more descriptive, e.g. `offsets`, `currentOffsets`, `latestOffsets`, etc. Same for `getOffset` below.